core/remote/change.js

```diff
@@ -120,8 +120,8 @@ function includeDescendant (parent /*: RemoteDirMove */, e /*: RemoteDescendantC
   delete e.descendantMoves
 }
 
-const addPath = (a /*: RemoteChange */) /*: ?string */ => isAdd(a) || isMove(a) || isRestore(a) ? a.doc.path : null
-const delPath = (a /*: RemoteChange */) /*: ?string */ => isDelete(a) ? a.doc.path : isMove(a) || isTrash(a) ? a.was.path : null
+const addPath = (a /*: RemoteChange */) /*: ?string */ => isAdd(a) || isMove(a) || isRestore(a) ? metadata.id(a.doc.path) : null
+const delPath = (a /*: RemoteChange */) /*: ?string */ => isDelete(a) ? metadata.id(a.doc.path) : isMove(a) || isTrash(a) ? metadata.id(a.was.path) : null
```

Comment:
I'm still not completely comfortable with our change because:

- `addPath()` and `delPath()` are actually returning ids
- Although `FOO/BAR` would be considered a child of `foo/` on a case-insensitive FS, this wouldn't be true on remote (i.e. `isChild*()` functions below now feel a bit weird)

Maybe renaming them would help, but I'm not sure...